recipes/cerbere/meta.yaml

```diff
@@ -0,0 +1,45 @@
+{% set name = "cerbere" %}
+{% set version = "V2.1.0" %}
+
+package: 
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://gitlab.ifremer.fr/cerbere/cerbere.git
+  git_tag: {{ version }}
```

Comment:
```suggestion
  url: https://gitlab.ifremer.fr/cerbere/cerbere/-/archive/V{{ version }}/cerbere-V{{ version }}.tar.gz
  sha256: f127d699086596c7434a8ab5b5562bd8f02973395d3e2d730e9dab01dbcfae51
```

Reply:
Same as for naiad-commons : With the use of poetry-dynamic-versioning the url does not work it seems to need the .git url. Is it really not possible to use the git_url ?
I have multiple package written that way sorry

Reply:
We discourage the use of `git_url` in lieu for a source distribution with a checksum for many reason:

1. smaller download
2. versioned tags can be changed, sdist with a checksum would fail if changed
3. 1-1 equivalence of the source distribution that could be sent to PyPI.

I'm not an expert on poetry, I use setuptools_scm, with scm there are options to record the version in the sdist of the tag. Maybe the same can be done with poetry?

Reply:
I am not an expert either on poetry I will check and see with the authors of the packages if it is possible. I'll get back to you ASAP. 
Thank you for your help !

Reply:
@xylar, @maresb, and @BastianZim, sorry for the ping but you are all listed as poetry maintainer too. Do you use that functionality? Any suggestions here?

Reply:
@Marie59, could you please be more specific here?

> Same as for naiad-commons : With the use of poetry-dynamic-versioning the url does not work it seems to need the .git url. 

What exactly goes wrong?

Reply:
@Marie59 and @ocefpaf, I'm afraid I don't use `poetry` (or `poetry-dynamic-versioning`) myself in my work.  I searched through the conda-forge feedstocks that use `poetry-dynamic-versioning` and I seems like either:
1. they are getting their source from PyPI and that includes a `setup.py` with the version number in addition to the `pyproject.toml` or
2. they are have `enabled = false` under `poetry-dynamic-versioning` in the `pyproject.toml`

I agree with @ocefpaf that it's not really safe enough to use the `git_url` because there's not a checksum.  I would say for now the best option (not a great one!) might be to make a `build.sh` and to use something like `sed` to replace the `0.0.0` in the source `pyproject.toml` with the version from conda environment variable, and disable `poetry-dynamic-versioning`. I'm afraid I can't think of a good alternative.

Reply:
> > Same as for naiad-commons : With the use of poetry-dynamic-versioning the url does not work it seems to need the .git url.
>
> What exactly goes wrong?

@maresb, this refers to another staged recipe:
https://github.com/conda-forge/staged-recipes/pull/21595

I presume what goes wrong is that `poetry-dynamic-versioning` attempts to use `git` to determine the version but since the source is not in a git repository, the version cannot be found.

Reply:
Have you tried setting an env variable?

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#use-environment-variables

https://github.com/mtkennerly/poetry-dynamic-versioning#environment-variables

Reply:
Yeah or just set the env variable in the meta.yml. 

Would probably be good to add a check to the test section where _output of version reported by software_ `= {{ version }}` and fail if not. 
That should catch possible failures.

Reply:
> Yeah or just set the env variable in the meta.yml.

Ah, I see.
```
  script_env:
   - POETRY_DYNAMIC_VERSIONING_BYPASS={{ version }}
  script: 
  - {{ PYTHON }} -m pip install . -vv
```

Reply:
Yes. I’m on my phone right now so wasn‘t able to test it but that should work.

Reply:
@Marie59, could you let us know if that works?